### ISSUE TITLE
inject uses result, instead of payload

### ISSUE THREAD

**Issue by jardakotesovec** [hapijs/bassmaster#29]
I have similar scenario to this:
```
{ "requests": [
    {"method": "get", "path": "/currentuser"},
    {"method": "get", "path": "/users/$0.id/profile"}
] }
```
I could not make it working. Finally I discovered that issue is that bassmaster is not using data.payload, but data.result. So it uses directly javascript object that was returned by first request. In my case I was returning object with toJSON method, so it worked perfectly fine without batching.

I was not sure if you would like PR for this, but I would think that it would make sense for consistency to use data.payload with JSON.parse(), to make sure you are dealing with correct response. Otherwise it can lead to unexpected results and long debugging.

**Comment by cadecairos** [comment]
@jardakotesovec Can you provide a use case for manually parsing the response data after inject VS using the pre-parsed result? I don't fully understand why it would be better.

**Comment by cadecairos** [comment]
Closing this issue, since it hasn't seen activity in 7 months.

@jardakotesovec, If you still have this issue with the latest version (just released v1.9.0), feel free to re-open and clarify the issue a bit more.
